comfyui-truss/model/model.py

```diff
@@ -0,0 +1,203 @@
+
+    def get_image(self, filename, subfolder, folder_type):
+        data = {"filename": filename, "subfolder": subfolder, "type": folder_type}
+        url_values = urllib.parse.urlencode(data)
+        with urllib.request.urlopen("http://{}/view?{}".format(self.server_address, url_values)) as response:
+            return response.read()
+
+    def get_history(self, prompt_id):
+        with urllib.request.urlopen("http://{}/history/{}".format(self.server_address, prompt_id)) as response:
+            return json.loads(response.read())
+
+    def get_images(self, ws, prompt):
+        prompt_id = self.queue_prompt(prompt)['prompt_id']
+        output_images = {}
+        while True:
+            out = ws.recv()
+            if isinstance(out, str):
+                message = json.loads(out)
+                if message['type'] == 'executing':
+                    data = message['data']
+                    if data['node'] is None and data['prompt_id'] == prompt_id:
+                        break  # Execution is done
+            else:
+                continue  # previews are binary data
+
+        history = self.get_history(prompt_id)[prompt_id]
+        for o in history['outputs']:
+            for node_id in history['outputs']:
+                node_output = history['outputs'][node_id]
+                if 'images' in node_output:
+                    images_output = []
+                    for image in node_output['images']:
+                        image_data = self.get_image(image['filename'], image['subfolder'], image['type'])
+                        images_output.append(image_data)
+                output_images[node_id] = images_output
+
+        return output_images
+
+    def pil_to_b64(self, pil_img):
+        buffered = BytesIO()
+        pil_img.save(buffered, format="PNG")
+        img_str = base64.b64encode(buffered.getvalue()).decode("utf-8")
+        return img_str
+
+    def predict(self, request: Dict) -> Dict:
+        if not self.ws:
+            self.ws = websocket.WebSocket()
+            self.ws.connect("ws://{}/ws?clientId={}".format(self.server_address, self.client_id))
+
+        json_workflow = request.pop("json_workflow")
+        template_values = request.pop("values")
+
+        tempfiles = []
+        for key, value in template_values.items():
+            if value.startswith("https://") or value.startswith("http://"):
+                if value[-1] == "/":
+                    value = value[:-1]
+                filename = value.split("/")[-1]
+                file_destination_path, file_object = self.download_tempfile(file_url=value, filename=filename)
+                tempfiles.append(file_object)
+                template_values[key] = file_destination_path
+
+        json_workflow = self.fill_template(json_workflow, template_values)
+
+        try:
+            images = self.get_images(self.ws, json_workflow)
+        except Exception as e:
+            print("SOMETHING BROKE!")
```

Comment:
you may want to break and return a meaningful error here. consider using `logging.error` or `logging.exception` instead of print